src/event_imp.rs

```diff
@@ -981,6 +986,450 @@ fn test_debug_ready() {
+    #[cfg(unix)]
+    pub fn is_hup(&self) -> bool {
+        (self.0 & HUP) != 0
+    }
+
+    /// Returns true if `Interests` contains error readiness
+    ///
+    /// **Note that only readable and writable readiness is guaranteed to be
+    /// supported on all platforms**. This means that `error` readiness should
+    /// be treated as a hint. For more details, see [readiness] in the poll
+    /// documentation.
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::error();
+    ///
+    /// assert!(interest.is_error());
+    /// ```
+    ///
+    /// [`Poll`]: ../struct.Poll.html
+    /// [readiness]: ../struct.Poll.html#readiness-operations
+    #[inline]
+    #[cfg(unix)]
+    pub fn is_error(&self) -> bool {
+        (self.0 & ERROR) != 0
+    }
+
+    /// Returns true if `Interests` contains AIO readiness
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::aio();
+    ///
+    /// assert!(interest.is_aio());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    #[cfg(any(
+        target_os = "dragonfly",
+        target_os = "freebsd",
+        target_os = "ios",
+        target_os = "macos"
+    ))]
+    #[inline]
+    pub fn is_aio(&self) -> bool {
+        (self.0 & AIO) != 0
+    }
+
+    /// Returns true if `Interests` contains LIO readiness
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::lio();
+    ///
+    /// assert!(interest.is_lio());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    #[cfg(any(target_os = "freebsd"))]
+    pub fn is_lio(&self) -> bool {
+        (self.0 & LIO) != 0
+    }
+
+    /// Returns true if `Interests` contains priority (`EPOLLPRI`) readiness
+    ///
+    /// See [`Poll`] for more documentation on polling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::Interests;
+    ///
+    /// let interest = Interests::priority();
+    ///
+    /// assert!(interest.is_priority());
+    /// ```
+    ///
+    /// [`Poll`]: struct.Poll.html
+    #[inline]
+    #[cfg(any(target_os = "linux", target_os = "android", target_os = "solaris"))]
+    pub fn is_priority(&self) -> bool {
+        (self.0 & PRI) != 0
+    }
+
+    /// Returns `Ready` contains `Interests` readiness
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mio::{Interests, Ready};
+    ///
+    /// let interest = Interests::readable();
+    ///
+    /// assert_eq!(Ready::readable(),interest.to_ready());
+    /// ```
+    pub fn to_ready(&self) -> Ready {
+        Ready(self.0)
+    }
+}
+
+impl<T: Into<Interests>> ops::BitOr<T> for Interests {
```

Comment:
Do we have anything that can be converted into `Interests`? Otherwise this can use `Interests` directly.

Reply:
I will use `Interests` directly.